h2/src/main/org/h2/engine/Database.java

```diff
@@ -201,7 +203,7 @@
     public Database(ConnectionInfo ci, String cipher) {
         String name = ci.getName();
         this.dbSettings = ci.getDbSettings();
-        this.reconnectCheckDelay = dbSettings.reconnectCheckDelay;
+        this.reconnectCheckDelayNs = TimeUnit.MILLISECONDS.toNanos(dbSettings.reconnectCheckDelay);
         this.compareMode = CompareMode.getInstance(null, 0);
```

Comment:
I have left all variables in the DbSettings as they are. They still use millisecond precision. So no renaming required from my point of view.